### PR TITLE
Assignment complete

### PR DESCRIPTION
**Pull request by etni**

### CODE REVIEW

File: src/AirMap.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    public class AirMap
+    {
+        private readonly Dictionary<string,List<string>> routes = new Dictionary<string, List<string>>();
+
+        /// <summary>
+        /// add a direct connection between two Airports
+        /// </summary>
+        /// <param name="start"></param>
+        /// <param name="destination"></param>
+        public void AddRoute(string start, string destination)
```
Comment (sungcheng) [inline src/AirMap.cs]: nit: source/destination sounds more appropriate.  If your going to use start, then end sounds more appropriate.

File: src/AirMap.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+        /// <summary>
+        /// Prints all the routes found between two Airports
+        /// </summary>
+        /// <param name="start"></param>
+        /// <param name="destination"></param>
+        public void PrintRoutes(string start, string destination)
+        {
+            Console.WriteLine($"From {start} to {destination}");
+            var result = FindRoutes(start, destination);
```
Comment (sungcheng) [inline src/AirMap.cs]: nit: var routes = FindRoutes(start, destination);

File: src/AirMap.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            else
+                routes.Add(start, new List<string>() { destination });
+        }
+
+        /// <summary>
+        /// Prints all the routes found between two Airports
+        /// </summary>
+        /// <param name="start"></param>
+        /// <param name="destination"></param>
+        public void PrintRoutes(string start, string destination)
```
Comment (sungcheng) [inline src/AirMap.cs]: same here, source instead of start

File: src/AirMap.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        {
+            Console.WriteLine($"From {start} to {destination}");
+            var result = FindRoutes(start, destination);
+            if (result.Count == 0)
+                Console.WriteLine("No Routes found");
+
+            foreach(var route in result)
```
Comment (sungcheng) [inline src/AirMap.cs]: space between foreach and paren

File: src/AirMap.cs [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                {
+                    found.Add(new LinkedList<string>(new[] {item}));
+                }
+                else
+                {
+                    if (currVisited.Contains(item)) continue;
+
+                    currVisited.Add(item);
+
+                    var results = FindRoutes(item, destination,currVisited);
```
Comment (sungcheng) [inline src/AirMap.cs]: add space after destination,

File: src/Program.cs [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                ("d", "a"),
+                ("b","c"),
+                ("c", "b"),
+                ("b","d"),
+                ("d","b")
+            };
+
+
+
+            foreach(var (from,to)  in routes)
```
Comment (sungcheng) [inline src/Program.cs]: spacing

File: src/Program.cs [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+                controller.AddRoute(from,to);
+
+            ProcessCommands();
+
+            //controller.PrintDirectDestinations();
```
Comment (sungcheng) [inline src/Program.cs]: rogue comments

File: src/Program.cs [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+                ("b","c"),
+                ("c", "b"),
+                ("b","d"),
+                ("d","b")
+            };
+
+
+
+            foreach(var (from,to)  in routes)
+                controller.AddRoute(from,to);
```
Comment (sungcheng) [inline src/Program.cs]: spacing

File: src/Airport.cs [hunk 9]
```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Generic;
+
+namespace AirPort
+{
+    public class TrafficController
```
Comment (sungcheng) [inline src/Airport.cs]: what is the difference between TrafficController and AirMap?  Looks like the same thing.  I would get rid of one of them.

File: src/Airport.cs [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+        /// </summary>
+        /// <param name="start"></param>
+        /// <param name="destination"></param>
+        /// <param name="visited">used internally by the function to keep track of where we have visited</param>
+        /// <returns></returns>
+        private List<LinkedList<string>> FindRoutes(string start, string destination, HashSet<string> visited = null)
+        {
+            var found = new List<LinkedList<string>>();
+
+            if (!routes.ContainsKey(start)) return found;
```
Comment (sungcheng) [inline src/Airport.cs]: stylistic nit:  I prefer return to be on the next line.

File: src/Airport.cs [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+
+                var currVisited  =  visited ?? new HashSet<string> {start};
+
+                if (item == destination)
+                {
+                    found.Add(new LinkedList<string>(new[] {item}));
+                }
+                else
+                {
+                    if (currVisited.Contains(item)) continue;
```
Comment (sungcheng) [inline src/Airport.cs]: stylistic nit: I prefer continue to be on the next line.

**Review by sungcheng — CHANGES_REQUESTED** [review]

File: src/AirMap.cs (hunk 1 above)
Comment (etni) [reply src/AirMap.cs]: I just used the names that they gave me for the assigment.

File: src/Airport.cs (hunk 9 above)
Comment (etni) [reply src/Airport.cs]: yeah, it was a glitch, it's an extra file.

**Comment by etni** [comment]
it's all fixed...  u mad?

**Comment by sungcheng** [comment]
@etni, almost addressed all the issues.  There is still formatting issues with program.cs

File: src/Program.cs (hunk 6 above)
Comment (etni) [reply src/Program.cs]: done

File: src/Program.cs (hunk 8 above)
Comment (etni) [reply src/Program.cs]: done

**Review by sungcheng — APPROVED** [review]
:shipit:  don’t forget to squash the commits

**Pull request merged**